proposals/3392/dataset_partitioning.md

```diff
@@ -0,0 +1,301 @@
+---
+Author: Paweł Leszczyński 
+Created: January tenth, 2025 
+---
+
+# OpenLineage dataset partitioning proposal
+
+Lineage metadata should contain information needed to answer the following questions:
+1. Is the dataset partitioned? How is it done?
+2. What partitions have been read by a particular run?
+3. What partitions have been updated by a particular run?
+
+There are two problems with answering (2). 
+First, it seems to be difficult to extract this information from running jobs. 
+Secondly, partition filtering is just a sub problem of describing a subset of the input dataset that has been used.
+
+## Proposed spec changes
+
+### Facet to specify a dataset is partitioned
+
+Create `PartitionDatasetFacet`
+```json
+{
+  "PartitionDatasetFacet": {
+    "allOf": [
+      {
+        "$ref": "https://openlineage.io/spec/2-0-2/OpenLineage.json#/$defs/DatasetFacet"
+      },
+      {
+        "type": "object",
+        "properties": {
+          "dimensions": {
+            "type": "array",
+            "items": {
+              "type": "object",
+              "properties": {
+                "fields": {
+                  "type": "array",
+                  "items": "string",
+                  "description": "Name of the fields used in partitioning."
+                },
+                "transform": {
+                  "type": "string"
+                }
+              }
+            }
+          },
+         "sqlClause": {
+            "type": "string",
+            "description": "Optional SQL clause definition like: 'PARTITIONED BY (days(ts), category)'."
+         }
```

Comment:
What is the purpose of storing only `PARTITIONED BY` part of table's DDL? I mean, if there is already a full DDL in another facet, could if be reused?

Reply:
How about renaming this into `partitionedBySqlClause` and omitting `PARTITIONED BY`? 
I think facets should be self-contained and some redundancy is Ok, even if there is another facet with a full DDL.

Reply:
So for PostgreSQL's `PARTITION BY RANGE (field)` it will be `RANGE(field)`?